app/components/profile_picture/image.tsx

```diff
@@ -60,25 +60,9 @@ const Image = ({author, forwardRef, iconSize, size, source, url}: Props) => {
         );
     }
 
-    let client: Client | undefined;
-
-    try {
-        client = NetworkManager.getClient(serverUrl);
-    } catch {
-        // handle below that the client is not set
-    }
-
-    if (author && client) {
-        let lastPictureUpdate = 0;
-        const isBot = ('isBot' in author) ? author.isBot : author.is_bot;
-        if (isBot) {
-            lastPictureUpdate = ('isBot' in author) ? author.props?.bot_last_icon_update : author.bot_last_icon_update || 0;
-        } else {
-            lastPictureUpdate = ('lastPictureUpdate' in author) ? author.lastPictureUpdate : author.last_picture_update || 0;
-        }
-
-        const pictureUrl = client.getProfilePictureUrl(author.id, lastPictureUpdate);
-        const imgSource = source ?? {uri: `${serverUrl}${pictureUrl}`};
+    if (author) {
+        const pictureUrl = buildProfileImageUrlFromUser(serverUrl, author);
```

Comment:
Is it intended that this behaves slightly differently before? If you can't get the client, it now passes an empty string into `buildAbsoluteUrl` instead of ending up that you render the `CompassIcon` further below

Reply:
Yes, the behavior is slightly different in theory, but in practice should be the same. All these components are in places where most probably the call to get client is not going to fail.

I was considering add another layer of refactoring to this PR to handle that case, but I felt it was not worth it.